cargo-miri/bin.rs

```diff
@@ -945,7 +945,13 @@ fn phase_runner(binary: &Path, binary_args: env::Args, phase: RunnerPhase) {
             // Drop this argument.
         } else if let Some(suffix) = arg.strip_prefix(json_flag) {
             assert!(suffix.starts_with('='));
-            // Drop this argument.
+            // This is how we pass through --color=always. We detect that Cargo is detecting rustc
+            // to emit the diagnostic structure that Cargo would consume from rustc to emit colored
+            // diagnostics, and ask rustc to emit them.
+            if arg.contains("diagnostic-rendered-ansi") {
```

Comment:
This is a comma-separated list... can you change this to split the list and do a `contains` on that, so that it doesn't match if another flag name contains this one as a substring?